app/assets/javascripts/components/option-select.js

```diff
@@ -243,7 +243,10 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
 
     // check whether this is hidden by progressive disclosure,
     // because height calculations won't work
-    if (this.$optionsContainer.offsetParent === null) {
+    // would use offsetParent === null but for IE10+
+    var parent = this.$optionSelect.parentElement
+    var parentIsHidden = !(parent.offsetWidth || parent.offsetHeight || parent.getClientRects().length)
+    if (parentIsHidden) {
```

Comment:
this seems twisted. to me, `offsetParent` being `null` could mean many things: the element itself or its parent have `display` set to `none`, or the element's `position` is set to `fixed` which makes the previous conditional statement unclear. now, I don't think we're making it any clear with this update.

can we instead look for the parent optionSelect and check if it's open or not? I see some `js-opened` `js-closed` classes that we can potentially check. may not be the best suggestion, but I'm looking at ways we can simplify this because it feels that we're moving from obscure to twisted.

Reply:
Totally with you on the obscureness here. Unfortunately we can't check the parent optionSelect - on mobile the search filters are inside a panel that appears/disappears like a modal, and the visibility is toggled way up the DOM.

Annoyingly IE10 supports the `offsetParent` null check, but it wasn't working in this test (although I wrote a quick test to check it and it should pass, but for some reason doesn't when run on the actual code). I replaced it with this because it seemed more accurate - it's actually checking that the parent element is hidden, rather than hidden or fixed, as you point out.

Really not sure what else to try with this - open to suggestions.